app/src/lib/stores/app-store.ts

```diff
@@ -2858,7 +2889,11 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return repository
     }
 
-    const repo = await this._checkoutBranch(repository, branch)
+    const repo = await this._checkoutBranch(
+      repository,
+      branch,
+      uncommittedChangesStrategy
```

Comment:
I want to call out this change because we had no way for `AppStore.createBranch` to pass its `uncommittedChangesStrategy` parameter to `AppStore.checkoutBranch`.

Without this parameter we'd always inherit the default behaviour `'AskForConfirmation'`, which means it would always show the "Switch Branch". But with this PR we want to bypass any other dialogs because we're assuming the user wants to take those changes with them, hence why I've added this parameter.